firecracker.go

```diff
@@ -183,16 +183,16 @@ func (f *Client) PutGuestDriveByID(ctx context.Context, driveID string, drive *m
 	return f.client.Operations.PutGuestDriveByID(params)
 }
 
-// PutGuestVsockByIDOpt is a functional option to be used for the
+// PutGuestVsockOpt is a functional option to be used for the
 // PutGuestVsockByID API in setting any additional optional fields.
 type PutGuestVsockByIDOpt func(*ops.PutGuestVsockByIDParams)
 
-// PutGuestVsockByID is a wrapper for the swagger generated client to make
+// PutGuestVsock is a wrapper for the swagger generated client to make
 // calling of the API easier.
-func (f *Client) PutGuestVsockByID(ctx context.Context, vsockID string, vsock *models.Vsock, opts ...PutGuestVsockByIDOpt) (*ops.PutGuestVsockByIDCreated, *ops.PutGuestVsockByIDNoContent, error) {
+func (f *Client) PutGuestVsockByID(ctx context.Context, vsock *models.Vsock, opts ...PutGuestVsockByIDOpt) (*ops.PutGuestVsockByIDNoContent, error) {
 	params := ops.NewPutGuestVsockByIDParams()
 	params.SetContext(ctx)
-	params.SetID(vsockID)
+	params.SetID(*vsock.VsockID)
```

Comment:
To prevent `nil` panics can we change this to `params.SetID(StringValue(vsock.VsockID)))`?